README.md
# Big Picture theme for Grav CMS

![Big Picture Theme screenshot](screenshot.jpg)

Big Picture theme is a port of [Big Picture](http://html5up.net/big-picture) by [HTML5 UP](http://html5up.net/).

## Demo

You can find a demo of the theme at [grav-theme-big-picture.flatfiledeveloper.com](https://grav-theme-big-picture.flatfiledeveloper.com).

## Features

* One page with contact form
* Fully responsive
* Built on HTML5/CSS3/[Skel](https://github.com/n33/skel)
* Font Awesome ready
* Released under [Creative Commons Attribution 3.0 License](http://html5up.net/license)

## Installation

### GPM Installation (Preferred)

The simplest way to install this theme is via the [Grav Package Manager (GPM)](http://learn.getgrav.org/advanced/grav-gpm) through your system's Terminal (also called the command line).  From the root of your Grav install type:

    bin/gpm install big-picture

This will install the Gateway theme into your `/user/themes` directory within Grav. Its files can be found under `/your/site/grav/user/themes/big-picture`.

### Manual Installation

To install this theme, just download the zip version of this repository and unzip it under `/your/site/grav/user/themes`. Then, rename the folder to `big-picture`.

You should now have all the theme files under

    /your/site/grav/user/themes/big-picture

>> NOTE: This theme is a modular component for Grav which requires the [Grav](http://github.com/getgrav/grav), [Error](https://github.com/getgrav/grav-theme-error), [Problems](https://github.com/getgrav/grav-plugin-problems), [Email](https://github.com/getgrav/grav-plugin-email) and [Form](https://github.com/getgrav/grav-plugin-form) plugins.

## Setup

If you want to set Big Picture as the default theme, you can do so by following these steps:

* Navigate to `/your/site/grav/user/config`.
* Open the **system.yaml** file.
* Change the `theme:` setting to `theme: big-picture`.
* Save your changes.
* Clear the Grav cache. The simplest way to do this is by going to the root Grav directory in Terminal and typing `bin/grav clear-cache`.

Once this is done, you should be able to see the new theme on the frontend. Keep in mind any customizations made to the previous theme will not be reflected as all of the theme and templating information is now being pulled from the **big-picture** folder.

## Configuration with Admin plugin

If you have [Admin plugin](https://github.com/getgrav/grav-plugin-admin) installed, you can fully manage the sub-pages via Admin plugin.

* For all of the sub-pages, there is a `Heading` option in `Content` tab which is used to define the heading of each page.
* `Background Image` option is available in `Content` tab for `intro` and `info` page file.

![](screenshot_1.jpg)

* There is an `Images` tab for `gallery` template, it is used to setup the image gallery.

![](screenshot_2.jpg)

* To manage the items in footer, you edit the theme's settings. Links with icons are on the left side of footer, on the right side are custom text.

![](screenshot_3.jpg)

* When you create a page with `default` template (file name is `default.md`), you can show a menu item linked to your home page at the top right corner of the page, you just need to enter the label of the menu item to `Home Menu Item` field at the bottom of `Content` tab.

![](screenshot_3.jpg)

## Navigation Buttons

To have navigation button for each section, the folder name of your home page must be `home`, in its modular.md file you need to setup the custom ordering of your modular pages. Here is an example:

```
title: Home
content:
    items: '@self.modular'
    order:
        by: default
        dir: asc
        custom:
            - _intro
            - _what-i-do
            - _who-i-am
            - _my-work
            - _contact
```

## Updating

### GPM Update (Preferred)

The simplest way to update this theme is via the [Grav Package Manager (GPM)](http://learn.getgrav.org/advanced/grav-gpm). You can do this with this by navigating to the root directory of your Grav install using your system's Terminal (also called command line) and typing the following:

    bin/gpm update big-picture

This command will check your Grav install to see if your Gateway theme is due for an update. If a newer release is found, you will be asked whether or not you wish to update. To continue, type `y` and hit enter. The theme will automatically update and clear Grav's cache.

### Manual Update

Manually updating Big Picture is pretty simple. Here is what you will need to do to get this done:

* Delete the `your/site/user/themes/big-picture` directory.
* Download the new version of the Big Picture theme from this repository.
* Unzip the zip file in `your/site/user/themes` and rename the resulting folder to `big-picture`.
* Clear the Grav cache. The simplest way to do this is by going to the root Grav directory in terminal and typing `bin/grav clear-cache`.

> Note: Any changes you have made to any of the files listed under this directory will also be removed and replaced by the new set. Any files located elsewhere (for example a YAML settings file placed in `user/config/themes`) will remain intact.


CHANGELOG.md
# v3.0.1
## 27/01/2023

1. [](#bugfix)
    * Fixed background issue.

# v3.0.0
## 27/01/2023

1. [](#improved)
    * Updated to the latest upstream version.

# v2.0.1
## 18/12/2022

1. [](#bugfix)
    * Fix blueprint for module info, intro (thanks to @Thaonguyen4488)

# v2.0.0
## 31/03/2022

1. [](#improved)
    * Updated to the latest upstream version.

# v1.4.3
## 11/04/2021

1. [](#bugfix)
    * Fixed Grav 1.7 incompatibility.

# v1.4.2
## 29/03/2021

1. [](#bugfix)
    * Added raw filter for Grav 1.7 compatibility.

# v1.4.1
## 06/05/2019

1. [](#bugfix)
    * Fixed issue when saving gallery modular page (thanks to @trangthaonguyen)

# v1.4.0
## 30/03/2018

1. [](#improved)
    * Added language file
    * Added ability to show home menu item in non-modular page (thanks to @trangthaonguyen)
2. [](#bugfix)
    * Fixed small issue in form's HTML (thanks to @der-trainer)

# v1.3.0
## 09/11/2017

1. [](#improved)
    * Added ability to select gallery's full image via image dropdown list
    * Removed gallery's thumbnail image
    * Added 2 options for gallery thumbnail's max width and max height
    * Gallery's thumbnail image is automatically resized from full image based on defined max width and max height values

# v1.2.0
## 05/11/2017

1. [](#improved)
    * Added ability to select image in page form

# v1.1.0
## 05/11/2017

1. [](#improved)
    * Added error template and ability to upload background image for error page
    * Added ability to select favicon

# v1.0.4
## 09/07/2017

1. [](#bugfix)
    * Improved link color
    * Improved JavaScript

# v1.0.3
## 15/09/2016

1. [](#bugfix)
    * Fixed mixed content issue
    * Fixed contact form for compatibility with new version of Form plugin

# v1.0.2
## 15/07/2016

1. [](#bugfix)
    * Removed hardcoded content in default.html.twig

# v1.0.1
## 03/06/2016

1. [](#bugfix)
    * Fixed in correct filenames.

# v1.0.0
## 02/06/2016

1. [](#new)
    * Theme started...


assets/js/main.js
/*
	Big Picture by HTML5 UP
	html5up.net | @n33co
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
*/

(function($) {

	skel.breakpoints({
		wide: '(max-width: 1920px)',
		normal: '(max-width: 1680px)',
		narrow: '(max-width: 1280px)',
		narrower: '(max-width: 1000px)',
		mobile: '(max-width: 736px)',
		mobilenarrow: '(max-width: 480px)',
	});

	$(function() {

		var	$window = $(window),
			$body = $('body'),
			$header = $('#header'),
			$all = $body.add($header);

		// Disable animations/transitions until the page has loaded.
			$body.addClass('is-loading');

			$window.on('load', function() {
				window.setTimeout(function() {
					$body.removeClass('is-loading');
				}, 0);
			});

		// Touch mode.
			skel.on('change', function() {

				if (skel.vars.mobile || skel.breakpoint('mobile').active)
					$body.addClass('is-touch');
				else
					$body.removeClass('is-touch');

			});

		// Fix: Placeholder polyfill.
			$('form').placeholder();

		// Prioritize "important" elements on mobile.
			skel.on('+mobile -mobile', function() {
				$.prioritize(
					'.important\\28 mobile\\29',
					skel.breakpoint('mobile').active
				);
			});

		// CSS polyfills (IE<9).
			if (skel.vars.IEVersion < 9)
				$(':last-child').addClass('last-child');

		// Gallery.
			$window.on('load', function() {
				$('.gallery').poptrox({
					baseZIndex: 10001,
					useBodyOverflow: false,
					usePopupEasyClose: false,
					overlayColor: '#1f2328',
					overlayOpacity: 0.65,
					usePopupDefaultStyling: false,
					usePopupCaption: true,
					popupLoaderText: '',
					windowMargin: (skel.breakpoint('mobile').active ? 5 : 50),
					usePopupNav: true
				});
			});

		// Section transitions.
			if (!skel.vars.mobile
			&&	skel.canUse('transition')) {

				var on = function() {

					// Generic sections.
						$('.main.style1')
							.scrollex({
								mode:		'middle',
								delay:		100,
								initialize:	function() { $(this).addClass('inactive'); },
								terminate:	function() { $(this).removeClass('inactive'); },
								enter:		function() { $(this).removeClass('inactive'); },
								leave:		function() { $(this).addClass('inactive'); }
							});

						$('.main.style2')
							.scrollex({
								mode:		'middle',
								delay:		100,
								initialize:	function() { $(this).addClass('inactive'); },
								terminate:	function() { $(this).removeClass('inactive'); },
								enter:		function() { $(this).removeClass('inactive'); },
								leave:		function() { $(this).addClass('inactive'); }
							});

					// Work.
						$('#work')
							.scrollex({
								top:		'40vh',
								bottom:		'30vh',
								delay:		50,
								initialize:	function() {

												var t = $(this);

												t.find('.row.images')
													.addClass('inactive');

											},
								terminate:	function() {

												var t = $(this);

												t.find('.row.images')
													.removeClass('inactive');

											},
								enter:		function() {

												var t = $(this),
													rows = t.find('.row.images'),
													length = rows.length,
													n = 0;

												rows.each(function() {
													var row = $(this);
													window.setTimeout(function() {
														row.removeClass('inactive');
													}, 100 * (length - n++));
												});

											},
								leave:		function(t) {

												var t = $(this),
													rows = t.find('.row.images'),
													length = rows.length,
													n = 0;

												rows.each(function() {
													var row = $(this);
													window.setTimeout(function() {
														row.addClass('inactive');
													}, 100 * (length - n++));
												});

											}
							});

					// Contact.
						$('#contact')
							.scrollex({
								top:		'50%',
								delay:		50,
								initialize:	function() { $(this).addClass('inactive'); },
								terminate:	function() { $(this).removeClass('inactive'); },
								enter:		function() { $(this).removeClass('inactive'); },
								leave:		function() { $(this).addClass('inactive'); }
							});

				};

				var off = function() {

					// Generic sections.
						$('.main.style1')
							.unscrollex();

						$('.main.style2')
							.unscrollex();

					// Work.
						$('#work')
							.unscrollex();

					// Contact.
						$('#contact')
							.unscrollex();

				};

				skel.on('change', function() {

					if (skel.breakpoint('mobile').active)
						(off)();
					else
						(on)();

				});

			}

		// Events.
			var resizeTimeout, resizeScrollTimeout;

			$window
				.resize(function() {

					// Disable animations/transitions.
						$body.addClass('is-resizing');

					window.clearTimeout(resizeTimeout);

					resizeTimeout = window.setTimeout(function() {

						// Update scrolly links.
							$('a[href^=\\#]').scrolly({
								speed: 1500,
								offset: $header.outerHeight() - 1
							});

						// Re-enable animations/transitions.
							window.setTimeout(function() {
								$body.removeClass('is-resizing');
								$window.trigger('scroll');
							}, 0);

					}, 100);

				})
				.load(function() {
					$window.trigger('resize');
				});

	});

})(jQuery);
